tests/modules/general/test-All_in_One_SEO_Pack_GetMainDescription.php

```diff
@@ -0,0 +1,253 @@
+<?php
+/**
+ * Testing All_in_One_SEO_Pack_Sitemap::get_main_description();
+ *
+ * Function:  Get's the AIOSEOP, post meta data, description from the post object (including taxonomies, attachments, etc)
+ *            Which is also used for the General module ( aioseop_class ), OpenGraph module, and Page Snippet.
+ * Expected:  Returns a (string) description depending on the $aioseop_options configuration with auto-gen, truncate,
+ *            and Never Shorten Length.
+ * Actual:    In some instances, the string isn't being truncated correctly.
+ * Reproduce: Set the _aioseop_description (maybe include OG module class), set the options,
+ *            and current screen to target post, front page, taxonomy (maybe more).
+ *
+ * @since 2.4.4.1
+ *
+ * @ticket 1395 Support longer meta descriptions
+ * @ticket 1468 broken aioseop_description filter
+ *
+ * @group All_in_One_SEO_Pack
+ * @group get_main_description
+ * @group post
+ * @group meta
+ * @group description
+ *
+ */
+
+require_once AIOSEOP_UNIT_TESTING_DIR . '/base/class-aioseop-test-base.php';
+
+/**
+ * Contains the test case scenario.
+ *
+ * Extending allows adding WP's Testing Unit; which extends to PHPUnit
+ *
+ * @package WP_UnitTestCase
+ */
+class All_in_One_SEO_Pack_Test_GetMainDescription extends AIOSEOP_Test_Base	{
```

Comment:
@EkoJR The class name is too clunky. Do we necessarily have to use "All_in_One_SEO_Pack" when we are obviously in the plugin folder?

Reply:
It is a Class name (All_in_One_SEO_Pack) testing a Method/Function (get_main_description). I was trying to make it descriptive to point to a specific Class->Method, and the test method is testing the return value's length.

Reply:
@EkoJR the method's name might change and this would render this naming convention useless. I think it needs to be simpler so that it indicates the module. Maybe use the classes already present and if they don't suffice, create a new one.

Reply:
In this case, I was also following WP Standards.

> Test class names should reflect the filepath, with underscores replacing directory separators and TitleCase replacing camelCase. Thus, the class Tests_Comment_GetCommentClass(), which contains tests for the get_comment_class() function, is located in tests/comment/getCommentClass.php.

https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes

Reply:
@michaeltorbert do you want to weigh in? We need to be able to follow a convention and if we follow WP conventions as they have been defined, we will end up having one class testing one particular function instead of having one class testing a variety of functions that interact.